### PR TITLE
Added class, subclass, and protocol fields

### PR DESCRIPTION
**Pull request by realivanjx**

### CODE REVIEW

**Comment by MelbourneDeveloper** [comment]
@realivanjx you're on the right track here. Unfortunately, ConnectedDeviceDefinition is shared across multiple device types, and these properties are USB specific. So, there will need to be something like a ConnectedUSBDeviceDefinitionBase class.

**Comment by realivanjx** [comment]
How about creating sub usb definition specific class like Stream has sub class FileStream etc?

**Comment by MelbourneDeveloper** [comment]
@realivanjx yes. ConnectedUSBDeviceDefinition will inherit from ConnectedDeviceDefinition. But, it must only be returned for USB devices - not Hid devices.

**Comment by realivanjx** [comment]
Yes that would be sufficient. Just return `ConnectedDeviceDefinition` from universal `DeviceManager` class (which will call device specific info query) then i would just cast it into `ConnectedUsbDeviceDefinition`.

That should work.

**Comment by MelbourneDeveloper** [comment]
@realivanjx if you can do that, there's a high chance I will accept the PR

**Comment by MelbourneDeveloper** [comment]
@realivanjx just bear in the mind that this will probably go in to 3.1. I'm about to release 3.0 in the next couple of days I think.

**Comment by realivanjx** [comment]
yes but can we fix the #99 first? do you have any idea whats happening there?

**Comment by MelbourneDeveloper** [comment]
@realivanjx #99 is scheduled for 3.1: https://github.com/MelbourneDeveloper/Device.Net/projects/10

> do you have any idea whats happening there?

It's probably as you said. The code is not getting called.

**Comment by realivanjx** [comment]
you sure to delay this for next release? this also explain why you dont get the buffer sizes from `ConnectedDeviceDefinition`.

**Comment by MelbourneDeveloper** [comment]
@realivanjx if you can fix it, it can go in to 3.0. But, as I said, this PR isn't finished yet. You need to split the class out in to ConnectedUSBDeviceDefinition. 

Is there some reason you need it in the 3.0 NuGet release?

**Comment by realivanjx** [comment]
no im just confused how to get interface handle and use it with `WindowsUsbDeviceFactory.GetDeviceDefinition()`. i dont actually need the nuget i can just use `.dll` files when fixed. im on a schedule now.

**Comment by MelbourneDeveloper** [comment]
@realivanjx please give ma couple of days. I will be able to investigate then. I cannot investigate properly right now.

**Comment by realivanjx** [comment]
okay

File: src/Device.Net - All.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.27703.2000
+# Visual Studio Version 16
```
Comment (MelbourneDeveloper) [inline src/Device.Net - All.sln]: No changes to projector solution files please.

File: src/Device.Net/ConnectedDeviceDefinitionBase.cs [hunk 2]
```diff
@@ -36,5 +36,11 @@ public abstract class ConnectedDeviceDefinitionBase : DeviceDefinitionBase
         /// The maximum size of data to be read from the device
         /// </summary>
         public int? ReadBufferSize { get; set; }
+
```
Comment (MelbourneDeveloper) [inline src/Device.Net/ConnectedDeviceDefinitionBase.cs]: These cannot be added to ConnectedDeviceDefinitionBase because not all device types support these.

File: src/Usb.Net/Windows/WindowsUsbDeviceFactory.cs [hunk 3]
```diff
@@ -41,7 +42,22 @@ public IDevice GetDevice(ConnectedDeviceDefinition deviceDefinition)
         #region Private Static Methods
         protected override ConnectedDeviceDefinition GetDeviceDefinition(string deviceId)
         {
-            return GetDeviceDefinitionFromWindowsDeviceId(deviceId, DeviceType.Usb, Logger);
+            // Getting device interface handle.
```
Comment (MelbourneDeveloper) [inline src/Usb.Net/Windows/WindowsUsbDeviceFactory.cs]: I think I understand what you are doing here.

Are you saying that we need to open a handle for each device to get the details you need?

File: src/testusb/App.config [hunk 4]
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
```
Comment (MelbourneDeveloper) [inline src/testusb/App.config]: Please submit a separate PR for extra apps.

File: src/Usb.Net/Windows/WindowsUsbDeviceFactory.cs (hunk 3 above)
Comment (realivanjx) [reply src/Usb.Net/Windows/WindowsUsbDeviceFactory.cs]: yes. previously your code doesnt even call the function to get the details i need so im trying to figure out how to get handle (i dont know if it is right) and call that function.

Comment (MelbourneDeveloper) [reply src/Usb.Net/Windows/WindowsUsbDeviceFactory.cs]: Understood

Comment (realivanjx) [reply src/Usb.Net/Windows/WindowsUsbDeviceFactory.cs]: is it right though to create handle like that?

Comment (MelbourneDeveloper) [reply src/Usb.Net/Windows/WindowsUsbDeviceFactory.cs]: I can't answer without doing research, but probably.

Comment (realivanjx) [reply src/Usb.Net/Windows/WindowsUsbDeviceFactory.cs]: do you have any doc abt `USB_DEVICE_DESCRIPTOR` buffer?

**Comment by realivanjx** [comment]
new findings. my changes may fixed the port max buffer size, manufacturer, product name, etc, but i just found that if i added this:

image

it will error `device handle no good. error code 5`.

**Comment by realivanjx** [comment]
i think it is because you call `WinUsb_Initialize` again inside `WindowsUsbInterfaceManager` class so that kinda duplicates the handles which cause this issue. thats also where i got the code to get default interface handle though.

**Comment by MelbourneDeveloper** [comment]
@realivanjx you're doing well, and I appreciate this. I will be able to look at it soon.

**Comment by MelbourneDeveloper** [comment]
@realivanjx how did you go with this. Did you redo this PR? It looks like it hasn't changed.

**Comment by realivanjx** [comment]
I havent been able to continue this pr bc i have another work to do
